kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ReplicaManagerImpl.java

```diff
@@ -84,15 +88,22 @@ public CompletableFuture<List<PartitionReplica>> searchReplicasByBrokerId(
               return KafkaFutures.toCompletableFuture(result.values().get(brokerId));
             })
         .thenCompose(
-            logDirs ->
-                CompletableFutures.allAsList(
-                    logDirs.values().stream()
-                        .flatMap(logDir -> logDir.replicaInfos.keySet().stream())
-                        .map(
-                            partition ->
-                                getReplica(
-                                    clusterId, partition.topic(), partition.partition(), brokerId))
-                        .collect(Collectors.toList())))
+            logDirs -> {
+              logDirs.forEach((key, value) -> log.debug("Describe log Dirs" + key + value));
+              CompletableFuture<List<Optional<PartitionReplica>>> res =
```

Comment:
Just return here, no need for the two step approach of making the variable and returning it.